common/config/src/lib.rs

```diff
@@ -103,8 +104,12 @@ pub struct JournaldConfig {
 }
 
 impl Config {
-    pub fn new() -> Result<Self, ConfigError> {
-        let argv_options = ArgumentOptions::from_args_with_all_env_vars();
+    pub fn new<I>(args: I) -> Result<Self, ConfigError>
```

Comment:
Will this change the existing behavior or just adding an additional way to pass configs?

Reply:
It doesn't change the behaviour, it just extracts the implicit call to std::env::args_os() from inside 

```rust
    ArgumentOptions::from_args()
```
and gives us the ability to pass in an arg list ourselves. (which we now explicitly get from std::env::args_os() in main.rs)